scripts/offline_data_processing.py

```diff
@@ -0,0 +1,257 @@
+            (item for item in parsed_output if hasattr(item, "num_datasets_shard")),
+            None,
+        )
+        num_datasets_shard = namespace_args.num_datasets_shard if namespace_args else 1
+
+        # If any of the arguments are None, raise an error
+        if None in [model_args, data_args, training_args]:
+            raise ValueError(
+                "One of the arguments is None. Please check the arguments passed."
+            )
+
+        logger.debug(
+            "Input args parsed:\n"
+            "  model_args: %s\n"
+            "  data_args: %s\n"
+            "  training_args: %s\n",
+            model_args,
+            data_args,
+            training_args,
+            num_datasets_shard,
+        )
+        training_args, logger = set_log_level(training_args, __name__)
+    except Exception as e:  # pylint: disable=broad-except
+        logger.error("Error parsing arguments: %s", traceback.format_exc())
+        write_termination_log(
+            f"Exception raised during argument parsing. This may be a problem with your input: {e}"
+        )
+        sys.exit(USER_ERROR_EXIT_CODE)
+
+    # Process dataset
+    try:
+        logger.info("Processing dataset with get_processed_dataset.")
+        formatted_train_dataset, formatted_validation_dataset = get_processed_dataset(
+            model_args=model_args,
+            data_args=data_args,
+            train_args=training_args,
+        )
+    except Exception as e:  # pylint: disable=broad-except
+        logger.error("Error processing dataset: %s", traceback.format_exc())
+        write_termination_log(
+            f"Exception raised during dataset processing.\
+            This may be a problem with your input: {e}"
+        )
+        sys.exit(USER_ERROR_EXIT_CODE)
+
+    # Save train dataset
+    train_dataset_dir = os.path.join(training_args.output_dir, "train_dataset")
+
+    logging.info(
+        "Trying to dump %d shards of train dataset at %s",
+        num_datasets_shard,
+        train_dataset_dir,
+    )
+    if formatted_train_dataset is not None:
+        os.makedirs(train_dataset_dir, exist_ok=True)
+        for shard_idx in range(num_datasets_shard):
+            shard = formatted_train_dataset.shard(
+                index=shard_idx, num_shards=num_datasets_shard
+            )
+            shard.to_parquet(f"{train_dataset_dir}/ds_{shard_idx:05d}.parquet")
+        logging.info(
+            "Dumped %d shards of train_dataset at %s",
+            num_datasets_shard,
+            train_dataset_dir,
+        )
+    else:
+        logging.warning("Train dataset is None. Not saving train dataset.")
+
+    # Save validation dataset
+    validation_dataset_dir = os.path.join(
+        training_args.output_dir, "validation_dataset"
+    )
+    logging.info(
+        "Trying to dump %d shards of validation dataset at %s",
+        num_datasets_shard,
+        validation_dataset_dir,
+    )
+    if formatted_validation_dataset is not None:
```

Comment:
could you move this and above code into a single place and call once for train dataset and other time for validation dataset so we avoid code duplication.